### PR TITLE
Add option to add deamon and update step function

### PR DESCRIPTION
**Pull request by domire8**
See commit messages for the changes in this PR. Review by commit

### CODE REVIEW

File: source/modulo_components/include/modulo_components/ComponentInterface.hpp [hunk 1]
```diff
@@ -601,6 +599,44 @@ inline state_representation::CartesianPose ComponentInterface<NodeT>::lookup_tra
...
+                                   "No publisher for predicate " << predicate.first << " found.");
+      return;
+    }
+    predicate_publishers_.at(predicate.first)->publish(msg);
+  }
+}
+
+template<class NodeT>
+inline void ComponentInterface<NodeT>::publish_outputs() {
+  for (const auto& publisher: this->outputs_) {
```
Comment (eeberhard) [inline source/modulo_components/include/modulo_components/ComponentInterface.hpp:618]: For this function and the daemon call-backs, when you need to access both elements of the pair, I would prefer to use structured bindings

```c++
for (const auto& [signal, publisher]: this->outputs_)
```

File: source/modulo_components/include/modulo_components/ComponentInterface.hpp [hunk 2]
```diff
@@ -241,6 +241,12 @@ class ComponentInterface : public NodeT {
 
   void step();
 
+  void publish_predicates();
```
Comment (eeberhard) [inline source/modulo_components/include/modulo_components/ComponentInterface.hpp:244]: docstrings for the new methods

**Review by eeberhard — COMMENTED** [review]
Very nice overall to tie together the step function. Minor comments only.

File: source/modulo_components/include/modulo_components/ComponentInterface.hpp [hunk 3]
```diff
@@ -290,56 +326,28 @@ ComponentInterface<NodeT>::ComponentInterface(
...
-    RCLCPP_DEBUG_STREAM(this->get_logger(), "Predicate " << name << " already exists, overwriting.");
-  } else {
-    this->predicate_publishers_.insert_or_assign(
-        name, this->template create_publisher<std_msgs::msg::Bool>(
-            this->generate_predicate_topic(name), 10
-        ));
-  }
-  this->predicates_.insert_or_assign(name, predicate);
+inline void ComponentInterface<NodeT>::step() {
+  this->publish_predicates();
```
Comment (domire8) [inline source/modulo_components/include/modulo_components/ComponentInterface.hpp:330]: As we just discussed, this is not correct. For a lifecycle component, publishing should only happen if the component is active

Comment (domire8) [reply source/modulo_components/include/modulo_components/ComponentInterface.hpp]: Should be good now

File: source/modulo_components/include/modulo_components/ComponentInterface.hpp [hunk 4]
```diff
@@ -174,6 +179,13 @@ class ComponentInterface : public NodeT {
       bool fixed_topic = false, const std::string& default_topic = ""
   );
 
+  /**
+   * @brief Function to daemonize the callback function given in input.
```
Comment (eeberhard) [inline source/modulo_components/include/modulo_components/ComponentInterface.hpp:183]: Just a suggestion to improve this docstring
```suggestion
   * @brief Add a periodic callback function.
   * @details The provided function is evaluated periodically at the component step period. 
```
And, as discussed, perhaps `add_periodic_callback` is better than `add_daemon` as a function name.

**Review by eeberhard — APPROVED** [review]

**Pull request merged**